lib/compat/wordpress-6.2/block-patterns.php

```diff
@@ -111,3 +111,182 @@ function gutenberg_register_core_block_patterns() {
 	}
 }
 add_action( 'init', 'gutenberg_register_core_block_patterns' );
+
+
+/**
+ * Block patterns registration from within a theme
+ *
+ * @package gutenberg
+ */
+
+/**
+ * Register any patterns that the active theme may provide under its
+ * `./patterns/` directory. Each pattern is defined as a PHP file and defines
+ * its metadata using plugin-style headers. The minimum required definition is:
+ *
+ *     /**
+ *      * Title: My Pattern
+ *      * Slug: my-theme/my-pattern
+ *      *
+ *
+ * The output of the PHP source corresponds to the content of the pattern, e.g.:
+ *
+ *     <main><p><?php echo "Hello"; ?></p></main>
+ *
+ * If applicable, this will collect from both parent and child theme.
+ *
+ * Other settable fields include:
+ *
+ *   - Description
+ *   - Viewport Width
+ *   - Categories       (comma-separated values)
+ *   - Keywords         (comma-separated values)
+ *   - Block Types      (comma-separated values)
+ *   - Post Types       (comma-separated values)
+ *   - Inserter         (yes/no)
+ *
+ * @since 6.0.0
+ * @access private
+ */
+function gutenberg_register_theme_block_patterns() {
+	$default_headers = array(
+		'title'         => 'Title',
+		'slug'          => 'Slug',
+		'description'   => 'Description',
+		'viewportWidth' => 'Viewport Width',
+		'categories'    => 'Categories',
+		'keywords'      => 'Keywords',
+		'blockTypes'    => 'Block Types',
+		'postTypes'     => 'Post Types',
+		'templateTypes' => 'Template Types',
+		'inserter'      => 'Inserter',
+	);
+
+	/*
+	 * Register patterns for the active theme. If the theme is a child theme,
+	 * let it override any patterns from the parent theme that shares the same slug.
+	 */
+	$themes     = array();
+	$stylesheet = get_stylesheet();
+	$template   = get_template();
+	if ( $stylesheet !== $template ) {
+		$themes[] = wp_get_theme( $stylesheet );
+	}
+	$themes[] = wp_get_theme( $template );
```

Comment:
```suggestion
	$wp_theme = wp_get_theme();
	if ( $wp_theme->parent() ) {
		$themes[] = $wp_theme->parent();
	}
	$themes[] = $wp_theme;
```

Reply:
It seems you already fixed this issue in a separate PR, thank you!